extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -130,6 +142,78 @@ export class AmpSlideScroll extends BaseCarousel {
   }
 
   /**
+   * Handles scroll on the slides container.
+   * @param {!Event} event Event object.
+   * @private
+   */
+  scrollHandler_(event) {
+    if (this.scrollTimeout_) {
+      timer.cancel(this.scrollTimeout_);
+    }
+    const currentScrollLeft = this.slidesContainer_./*REVIEW*/scrollLeft;
+    if (currentScrollLeft != this.previousScrollLeft_ &&
+        !this.hasNativeSnapPoints_) {
+      //Handle custom scroll here.
+    }
+    this.scrollTimeout_ = timer.delay(() => {
+      if (this.snappingInProgress_) {
+        return;
+      }
+      if (this.hasNativeSnapPoints_) {
+        this.updateOnScroll_(currentScrollLeft);
+      }
+    }, 100);
+    this.previousScrollLeft_ = currentScrollLeft;
+  }
+
+
+  /**
+   * Updates to the right state of the new index on scroll.
+   * @param {number} currentScrollLeft scrollLeft value of the slides container.
+   */
+  updateOnScroll_(currentScrollLeft) {
+    this.snappingInProgress_ = true;
+    // This can be only 0, 1 or 2, since only a max of 3 slides are shown at
+    // a time.
+    const scrolledSlideIndex = Math.round(currentScrollLeft / this.slideWidth_);
+    // Update value can be -1, 0 or 1 depending upon the index of the current
+    // shown slide.
+    let updateValue = 0;
+
+    const hasPrev_ = this.hasPrev();
+    const hasNext_ = this.hasNext();
+
+    if (hasPrev_ && hasNext_) {
+      updateValue = scrolledSlideIndex - 1;
+    } else if (hasNext_) {
+      // Has next and does not have a prev. (slideIndex 0)
+      updateValue = scrolledSlideIndex;
+    } else if (hasPrev_) {
+      // Has prev and no next slide (last slide)
+      updateValue = scrolledSlideIndex - 1;
+    }
+
+    let newIndex = this.slideIndex_ + updateValue;
+
+    if (this.hasLooping_) {
+      newIndex = (newIndex < 0) ? this.noOfSlides_ - 1 :
+          (newIndex >= this.noOfSlides_) ? 0 : newIndex;
+    } else {
+      newIndex = (newIndex < 0) ? 0 :
+          (newIndex >= this.noOfSlides_) ? this.noOfSlides_ - 1 : newIndex;
+    }
+    this.vsync_.mutatePromise(() => {
+      this.slidesContainer_.classList.add('no-scroll');
+      this.showSlide_(newIndex);
+    }).then(() => {
```

Comment:
I'm pretty sure that if you delete line `208` change `205` to just `mutate` it does the same thing. Or am I missing something?